chore: assign code owners to frontend and backend modules in CODEOWNERS

laipz8200 opened the pull request:
## Summary

- Add owner assignments to frontend and backend modules including Workflow Engine, RAG, Plugins, Billing, Enterprise, and Database Migrations

Closes #28712

laipz8200 committed:
chore: assign code owners to backend modules in CODEOWNERS

- Add owner assignments to Workflow Engine (@laipz8200 @QuantumGhost)
- Add new Workflow Nodes section for Agent/Iteration/Loop/LLM (@Novice)
- Add owner assignments to Workflow services (@QuantumGhost)
- Add owner assignments to RAG module (@JohnJyong)
- Add owner assignments to Plugins (@Mairuis @Yeuoly)
- Separate Billing from Enterprise (@hj24 @zyssyz123)
- Add owner assignments to Enterprise (@GarfieldDai @GareArc)
- Add new Database Migrations section (@snakevash @laipz8200)
- Standardize frontend comments with "Frontend - " prefix

Closes #28712

laipz8200 committed:
fix: reorder CODEOWNERS rules to ensure specific patterns take precedence

Move general api/core/workflow/ rule before more specific subdirectory
rules (graph_engine/, nodes/, etc.) so that the last matching pattern
correctly assigns the intended owners.

laipz8200 committed:
chore: add @Stream29 to plugin module owners

laipz8200 committed:
chore: remove redundant workflow frontend rules

Remove web/app/components/workflow-app/components/workflow-main.tsx and
web/app/components/workflow/store/ rules as they are already covered by
the more general web/app/components/workflow/ and web/app/components/workflow-app/
patterns with the same owners.

laipz8200 commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,220 @@
+web/app/components/workflow-app/components/workflow-main.tsx @iamjoel @zxhlyh
+web/app/components/workflow/store/ @iamjoel @zxhlyh
+web/app/components/rag-pipeline/components/rag-pipeline-main.tsx @iamjoel @zxhlyh
+web/app/components/rag-pipeline/store/ @iamjoel @zxhlyh
```
fixed

laipz8200 commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,220 @@
+api/tasks/enable_segments_to_index_task.py @JohnJyong
+api/tasks/clean_dataset_task.py @JohnJyong
+api/tasks/deal_dataset_index_update_task.py @JohnJyong
+api/tasks/deal_dataset_vector_index_task.py @JohnJyong
```
We might reorganize the file directory later, so using wildcards is not a good practice.

laipz8200 committed:
chore: add @Mairuis and @Yeuoly to trigger/schedule/webhook and async_workflow modules

laipz8200 commented:
/gemini review

QuantumGhost commented:
I suggest considering using teams for `CODEOWNERS` instead of assigning to specific individuals, as described in the documentation [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file:~:text=Teams%20can%20be%20specified%20as%20code%20owners%20as%20well.).

This would reduce the maintenance effort for `CODEOWNERS` when team members change.

[This doc](https://docs.github.com/en/organizations/organizing-members-into-teams/about-teams#nested-teams) also mentions that GitHub supports nested teams. However, I wasn't able to set it up properly in my personal organization, so I'm not sure if it's supported by our current GitHub Plan.

laipz8200 committed:
fix: reorder frontend RAG pipeline rules for correct ownership precedence

Consolidate rag-pipeline rules under 'Frontend - RAG - Orchestration' section
with general rule first and specific rules after to ensure correct ownership:
- web/app/components/rag-pipeline/ -> @iamjoel @WTW0313 (general)
- web/app/components/rag-pipeline/components/rag-pipeline-main.tsx -> @iamjoel @zxhlyh (specific)
- web/app/components/rag-pipeline/store/ -> @iamjoel @zxhlyh (specific)

laipz8200 committed:
chore: remove redundant backend workflow rules

Remove specific workflow rules that duplicate ownership already covered by
the general api/ rule. The api/ @QuantumGhost rule serves as the default
owner for all backend files, and more specific rules override it where needed.
